### PR TITLE
Move JNI part of bindings from this repo to mapbox-gl-native-android.

### PR DESCRIPTION
**Pull request by tarigo**
Resolves mapbox/mapbox-gl-native-android#202.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -566,7 +535,7 @@ jobs:
     steps:
       - run:
           name: Clone mapbox-gl-native-android in working directory
-          command: git clone https://github.com/mapbox/mapbox-gl-native-android.git .
+          command: git clone -b tarigo_pull_in_jni_bindings https://github.com/mapbox/mapbox-gl-native-android.git .
```
Comment (tarigo) [inline circle.yml]: This temporary workaround to get CI green, should be reverted back to master after mapbox/mapbox-gl-native-android#301 is merged.

Comment (tobrun) [reply circle.yml:538]: It's fine to merge the API breakage bot with a failing result, else it introduces a circular dependency and you won't be able to merge any of the two PRs. I prefer merging this PR first and updating downstream PR to master of gl-native.

Comment (tarigo) [reply circle.yml]: Reverted back to master.

File: circle.yml [hunk 2]
```diff
@@ -15,41 +15,6 @@ workflows:
...
-          config_params: '-G Ninja -DCMAKE_INSTALL_PREFIX=/tmp/workspace/install/$CIRCLE_JOB -DCMAKE_TOOLCHAIN_FILE=/opt/android/ndk-bundle/build/cmake/android.toolchain.cmake -DANDROID_CCACHE=/usr/bin/ccache -DANDROID_ABI=armeabi-v7a'
-          install: true
-      - build-template:
-          name: android-arm64-v8a-release
-          executor_name: ubuntu-disco
-          target_is_android: true
-          requires:
-            - sanity-checks
-          config_params: '-G Ninja -DCMAKE_INSTALL_PREFIX=/tmp/workspace/install/$CIRCLE_JOB -DCMAKE_TOOLCHAIN_FILE=/opt/android/ndk-bundle/build/cmake/android.toolchain.cmake -DANDROID_CCACHE=/usr/bin/ccache -DANDROID_ABI=arm64-v8a'
-          build_params: '--target libmapbox-gl.so'
```
Comment (tmpsantos) [inline circle.yml:33]: I'm concerned about this, we won't measure the binary size anymore if we do this. I'm testing a version of your PR here that keeps these android bots but will measure the size of the test runner at least.

Comment (tmpsantos) [reply circle.yml:33]: As agreed, we will bring this back and measure the size of the render-test-runner.so

**Pull request merged**